packages/oauth/oauth-types/src/oauth-protected-resource-metadata.ts

```diff
@@ -10,8 +11,20 @@ export const oauthProtectedResourceMetadataSchema = z.object({
    * REQUIRED. The protected resource's resource identifier, which is a URL that
    * uses the https scheme and has no query or fragment components. Using these
    * well-known resources is described in Section 3.
+   *
+   * @note This schema allows non https URLs for testing & development purposes.
+   * Make sure to validate the URL before using it in a production environment.
    */
-  resource: z.string().url(),
+  resource: webUriSchema
+    // .refine((url): url is `https://${string}` => url.startsWith('https://'), {
+    //   message: 'Resource URL must use the https scheme',
+    // })
```

Comment:
```suggestion
```

Reply:
This may have been left on purpose with the added note, but it's not obvious that the note is connected to it.  So it ends-up reading as dead code or code that was mistakenly commented-out.  I think the note on its own is sufficient 👍